### DISCUSSION TITLE
Auth-UI first load time

### DISCUSSION THREAD

**Discussion by strangecarr** [supabase/supabase#9437] [Questions] [answered]
Hi, I'm using the Auth-UI widget, it works fine for auth and I've customised it to my app's style. However, when I navigate from my landing page to the page using the widget, the first time it takes about 5 seconds to load. Then on subsequent loads it is lightning quick. Has anyone else experienced this and any advice on how to speed up the initial page load? I'm using next.js.

**Comment by Sukikiroi** [comment, marked as answer]
Hello,
Are you running in dev or production mode? In dev mode Next only compiles the page you are viewing so it will appear slower when navigating between routes. Production mode should be fine.

**Reply by strangecarr** [reply]
yes just dev mode at the moment, will see when switch to production. also just updated TailwindCSS mode to JIT so will see if that makes a difference. thanks a lot

**Reply by Sukikiroi** [reply]
Can you please mark my answer if u want ofc 
thank you
